source/extensions/filters/http/ext_proc/client_impl.cc

```diff
@@ -65,7 +65,11 @@ bool ExternalProcessorStreamImpl::close() {
 }
 
 void ExternalProcessorStreamImpl::onReceiveMessage(ProcessingResponsePtr&& response) {
-  callbacks_.onReceiveMessage(std::move(response));
+  if (!callbacks_.has_value()) {
+    ENVOY_LOG(debug, "Underlying filter object has been destroyed.");
```

Comment:
This doesn't make sense early in the lifecycle when the callbacks haven't yet been set.

Reply:
A few things to clarify:
(1) This happens after callbacks are set.  Callbacks are set [here](https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/ext_proc/ext_proc.cc#L299-L300) at the start of gRPC stream, while here is the place where server return the response.

(2) Please note this `callbacks_` is a reference to filter object and is set [here](https://github.com/envoyproxy/envoy/blob/main/source/extensions/filters/http/ext_proc/ext_proc.cc#L299-L300). I didn't change the variable naming but it might be a bit confusing ?

(3)This check is needed for deferred closure mode: When filter is being destroyed. the response from gRPC stream (which outlives the filter instance) could come back later asynchronously. At that time, the filter instance reference (i.e., callbacks_) becomes dangling reference that can not be accessed anymore

Reply:
OK, fair.